taigit/src/libraries/GitHub/src/numCommitsFromUser.ts

```diff
@@ -9,14 +9,30 @@ import axios from 'axios';
  */
 export async function
 getNumCommitsFromUser(owner: string, repo: string, author: string,
-    auth: string){
+                      auth: string){
     try{
+        let commits = 0
         var config = {
             headers: {'Authorization': "Bearer " + auth}
         }
         const userRepos = await axios.get("https://api.github.com/repos/" + owner +
             "/" + repo + "/commits" + "?author=" + author, config);
-        return userRepos.data.length;
+        if(userRepos.headers.hasOwnProperty("link")){
+            let last = userRepos.headers["link"].split(',');
+            let total_pages_str = last[1]
+            let total_pages = total_pages_str.substring(total_pages_str.lastIndexOf("page") + 5, total_pages_str.lastIndexOf(">"))
+            total_pages = Number(total_pages)
+            for (var j = 1; j <= total_pages; j++) {
+                const innerpulls = await axios.get("https://api.github.com/repos/" + owner +
```

Comment:
You have a link to the last page and you have the page size; why are you making all these extra network requests?